src/framework/components/element/text-element.js

```diff
@@ -342,7 +342,10 @@ class TextElement {
         if (this._enableMarkup) {
             const results = Markup.evaluate(this._symbols);
             this._symbols = results.symbols;
-            tags = results.tags;
+            // NOTE: if results.tags is null, we assign [] to increase
+            // probability of batching. So, if a user want to use as less
+            // WebGL buffers memory as possible they can just disable markups.
+            tags = results.tags || [];
```

Comment:
It is a suggestion that increases size of buffers if enableMarkup is true, but it also increases probability of batching.